spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/GcpCloudSqlAutoConfiguration.java

```diff
@@ -131,34 +130,36 @@ public CloudSqlJdbcInfoProvider cloudSqlJdbcInfoProvider(SQLAdmin sqlAdmin,
 	}
 
 	@Bean
-	@ConditionalOnMissingBean(DataSource.class)
-	public DataSource cloudSqlDataSource(CloudSqlJdbcInfoProvider cloudSqlJdbcInfoProvider)
-			throws GeneralSecurityException, IOException, ClassNotFoundException {
-		HikariConfig hikariConfig = new HikariConfig();
-		hikariConfig.setDriverClassName(cloudSqlJdbcInfoProvider.getJdbcDriverClass());
-		hikariConfig.setJdbcUrl(cloudSqlJdbcInfoProvider.getJdbcUrl());
-		hikariConfig.setUsername(this.gcpCloudSqlProperties.getUserName());
-		hikariConfig.setPassword(this.gcpCloudSqlProperties.getPassword());
-		// Setting this is useful to disable connection initialization. Especially in unit tests.
-		hikariConfig.setInitializationFailFast(this.gcpCloudSqlProperties.isInitFailFast());
-
-		return new HikariDataSource(hikariConfig);
+	@Primary
+	public DataSourceProperties dataSourceProperties(DataSourceProperties properties,
+			CloudSqlJdbcInfoProvider cloudSqlJdbcInfoProvider) {
+		if (StringUtils.isEmpty(properties.getUsername())) {
+			properties.setUsername("root");
```

Comment:
Would be nice to set an empty password as well, unless `getPassword()` works well with `null`.